src/app/tab/time/time.component.ts

```diff
@@ -45,6 +50,72 @@ export class TabTimeComponent implements OnInit {
     ) {
       this.shared.time = this.constructTitleTime();
     }
+    if (this.clock.binary.enabled) {
+
+      var hourList = this.hoursList.nativeElement.children;
+      var minuteList = this.minutesList.nativeElement.children;
+      var secondsList = this.secondsList.nativeElement.children;
+
+      var seconds = parseInt(this.getSecond(this.clock.timezone, 1) + this.getSecond(this.clock.timezone, 2));
+      var minutes = parseInt(this.getMinute(this.clock.timezone, 1) + this.getMinute(this.clock.timezone, 2));
+      var hours = parseInt(this.getHour(this.clock.timezone, this.clock.twentyFour, this.clock.twoDigit, 1) + this.getHour(this.clock.timezone, this.clock.twentyFour, this.clock.twoDigit, 2));
+
+      if (seconds == 0) {
+        this.resetdabs(secondsList);
+        if (minutes == 0) {
+          this.resetdabs(minuteList);
+          if (hours > 23) {
+            this.resetdabs(hourList);
+          }
+        }
+      }
+      this.UpdateBinaryClock(hourList, hours);
+      this.UpdateBinaryClock(minuteList, minutes);
+      this.UpdateBinaryClock(secondsList, seconds);
+    }
+  }
+
+  setActiveDabs(childrenList, v) {
+    var val = this.toBinary(parseInt(v))
+    this.resetChildren(childrenList);
+    for (var i = 0; i <= val.length - 1; i++) {
+      if (parseInt(val[(val.length - 1) - i]) > 0) {
+        childrenList[(childrenList.length - 1) - i].style.opacity ='100%';
+        if (this.clock.binary.dim) {
+          childrenList[(childrenList.length - 1) - i].style.opacity ='60%';
+          childrenList[(childrenList.length - 1) - i].classList.add('dim')
+        }
+      }
+    }
+  }
+
+  UpdateBinaryClock(childrenList, v) {
+    this.resetdabs(childrenList);
+    for (var s = 0; s <= v.toString().length - 1; s++) {
+      var childrensList = childrenList[(childrenList.length - 1) - s].children;
+      this.setActiveDabs(childrensList, parseInt(v.toString()[(v.toString().length - 1) - s]))
+    }
+  }
+
+  resetdabs = (childrenList) => {
+    for (var c = 0; c <= childrenList.length - 1; c++) {
+      this.resetChildren(childrenList[(childrenList.length - 1) - c].children);
+    }
+  }
+
+  resetChildren(childrenList) {
+    for (var i = childrenList.length - 1; i >= 0; i--) {
+      childrenList[(childrenList.length - 1) - i].style.opacity ='50%';
+      childrenList[(childrenList.length - 1) - i].style.backgroundColor =this.settings.config.design.foreground;
+      if (this.clock.binary.dim) {
+        childrenList[(childrenList.length - 1) - i].style.opacity ='20%';
+        childrenList[i].classList.add('dim')
+      }
+    }
+  }
+  
+  toBinary(v) {
+    return (v >>> 0).toString(2);
```

Comment:
Not a very common thing here, could you add some code comments to explain what is happening?

Reply:
this code was a bit over complicated and catered for negative numbers. I have simplified it. 
[Here ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Unsigned_right_shift)is more info about the right shift operator.